src/inmanta/execute/runtime.py

```diff
@@ -308,6 +309,49 @@ def get_dataflow_node(self) -> dataflow.AssignableNodeReference:
         return self._node
 
 
+# TODO: review implementation
+# TODO: double check typing
+class FixedCountVariable(ResultVariable[Union[T, list[T]]]):
```

Comment:
This is unique in that it doesn't behave as a set, but as a list?

Reply:
as it is now, it also breaks propagation of gradual execution

Reply:
I dropped this, replacing it with the `_results` attribute in the helper.